telemetry/util/heka_message.py

```diff
@@ -112,7 +175,7 @@ def read_one_record(input_stream, raw=False, verbose=False, strict=False, try_sn
     (header_length,) = struct.unpack('<B', header_length_raw)
 
     header_raw = input_stream.read(header_length)
-    if header_raw == '':
+    if header_length > 0 and header_raw == '': # no more data to read
```

Comment:
If the body is empty, then reading it will always produce a blank string and return None as a result. If `unpack` receives None, then it assumes the stream ended, and stops reading. So a stream with a message with no body can stop the rest of the stream from being read.

Parsing it like a normal message lets us continue reading messages.